lldb/source/Plugins/ExpressionParser/Swift/SwiftUserExpression.cpp

```diff
@@ -693,6 +694,69 @@ SwiftUserExpression::GetTextAndSetExpressionParser(
   return parse_result;
 }
 
+/// If `sc` represents a "closure-like" function according to `lang`, and
+/// `var_name` can be found in a parent context, create a diagnostic
+/// explaining that this variable is available but not captured by the closure.
+static std::string
+CreateVarInParentScopeDiagnostic(StringRef var_name,
+                                 StringRef parent_func_name) {
+  return llvm::formatv("Current frame is a closure.\nA variable named '{0}' "
+                       "exists in function '{1}', but it "
+                       "was not captured.\nHint: the variable may be available "
+                       "in a parent frame.",
```

Comment:
Is the hint contradictory? The message says "A variable named <v> exists in function <f>" – doesn't that mean a variable is guaranteed to be available in a parent frame?

Reply:
No, the function where a closure is declared is not necessarily a parent frame of the frame where the closure was called

Reply:
oh I see my misunderstanding, the diagnostic doesn't attempt check the stack for the function which has the found variable, which I assumed it would. I had pictured this as a runtime check, but it's more of a static check?

This ties back to my other question https://github.com/swiftlang/llvm-project/pull/10772#issuecomment-2940823331: In cases where the found function is on the stack (ex: call chains into a non-escaping closures in Swift), then should lldb print that variable?

Reply:
The misunderstanding I've had might not be unique to me. What do you think of:

> A variable named '{0}' existed where the current closure was defined (function '{1}'), but it was not captured. If '{1}' is in the call stack, select that frame and try again.

Reply:
I personally think either version works, but IIRC @adrian-prantl had some thoughts on being more verbose?